augur/filter/__init__.py

```diff
@@ -18,8 +18,8 @@ def register_arguments(parser):
     input_group.add_argument('--sequences', '-s', help="sequences in FASTA or VCF format")
     input_group.add_argument('--sequence-index', help="sequence composition report generated by augur index. If not provided, an index will be created on the fly.")
     input_group.add_argument('--metadata-chunk-size', type=int, default=100000, help="maximum number of metadata records to read into memory at a time. Increasing this number can speed up filtering at the cost of more memory used.")
-    input_group.add_argument('--metadata-id-columns', default=VALID_ID_COLUMNS, nargs="+", help="names of valid metadata columns containing identifier information.")
-    input_group.add_argument('--metadata-delimiters', default=DEFAULT_DELIMITERS, nargs="+", help="delimiters to accept when reading a metadata file.")
+    input_group.add_argument('--metadata-id-columns', default=VALID_ID_COLUMNS, nargs="+", help="names of valid metadata columns containing identifier information. Only one ID column will be inferred.")
+    input_group.add_argument('--metadata-delimiters', default=DEFAULT_DELIMITERS, nargs="+", help="delimiters to accept when reading a metadata file. Only one delimiter will be inferred.")
```

Comment:
_non-blocking_

+1 for this clarification!

For id columns, it might also be good to clarify that the first column that exists will be used. (right?)

(For delimiters, I think it's more complex than that.)

Reply:
Push this for ID columns: c0a9ddb9e132a89909234c003b501fe34111708f (yes, this logic lives in `augur.io.metadata.read_metadata()` and is documented there).

For delimiters, the logic lives in [`csv.Sniffer._guess_quote_and_delimiter()`](https://github.com/python/cpython/blob/b57105ae33e1f61e6bdf0eec45c4135d067b9b22/Lib/csv.py#L209) and is not documented. A quick look at the source code makes me think that it's whatever is first detected.

Reply:
:+1: 

FWIW, I earlier ended up looking at the code too for delimiter sniffing… and it's not first detected I think.

1. It first tries to guess that the characters next to any double- or single-quoted fields are delimiters ([ref](https://github.com/python/cpython/blob/b57105ae33e1f61e6bdf0eec45c4135d067b9b22/Lib/csv.py#L185-L186)). It picks the putative delim by which occurs the most (next to a quote) ([ref](https://github.com/python/cpython/blob/b57105ae33e1f61e6bdf0eec45c4135d067b9b22/Lib/csv.py#L259-L260)).
2. If that fails to find a delim (e.g. there are no quoted fields), then it does [frequency analysis](https://github.com/python/cpython/blob/b57105ae33e1f61e6bdf0eec45c4135d067b9b22/Lib/csv.py#L286-L302).